### PR TITLE
Enable snapshots

### PR DESCRIPTION
**Pull request by freshe4qa**
Enable snapshots for testnet to enable validators to more quickly deploy nodes/run experiments

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

**Comment by freshe4qa** [comment]
Try using a newer version of "go".

**Review by evan-forbes — COMMENTED** [review]
thanks for this!!

I believe all we have to do is to fix the linter before we merge, but I have yet to test this change.

**Comment by freshe4qa** [comment]
Okey, cool

File: cmd/celestia-appd/cmd/root.go [hunk 1]
```diff
@@ -251,6 +265,9 @@ func NewAppServer(logger log.Logger, db dbm.DB, traceStore io.Writer, appOpts se
 		baseapp.SetInterBlockCache(cache),
 		baseapp.SetTrace(cast.ToBool(appOpts.Get(server.FlagTrace))),
 		baseapp.SetIndexEvents(cast.ToStringSlice(appOpts.Get(server.FlagIndexEvents))),
+		baseapp.SetSnapshotStore(snapshotStore),
+		baseapp.SetSnapshotInterval(cast.ToUint64(appOpts.Get(server.FlagStateSyncSnapshotInterval))),
+                baseapp.SetSnapshotKeepRecent(cast.ToUint32(appOpts.Get(server.FlagStateSyncSnapshotKeepRecent))),
```
Comment (cmwaters) [inline cmd/celestia-appd/cmd/root.go:270]:
```suggestion
		baseapp.SetSnapshot(snapshotStore, snapshottypes.NewSnapshotOptions(cast.ToUint64(appOpts.Get(server.FlagStateSyncSnapshotInterval)), cast.ToUint32(appOpts.Get(server.FlagStateSyncSnapshotKeepRecent)))
```
I think my formatting is off but I believe this is all one function now.

**Review by cmwaters — CHANGES_REQUESTED** [review]
Thanks @freshe4qa for the PR. Just needs a touch up and then we can merge it

**Comment by freshe4qa** [comment]
I corrected it, check.

**Comment by freshe4qa** [comment]
I think we need to work on fixing the "lint".

**Comment by cmwaters** [comment]
Yeah just did that now

**Comment by freshe4qa** [comment]
Okey, cool.

**Review by cmwaters — APPROVED** [review]
Looks like everything is passing. Thanks for the PR

**Pull request merged**